include/libtorrent/session.hpp

```diff
@@ -184,14 +184,22 @@ namespace aux {
 		// In order to avoid a race condition between starting the session and
 		// configuring it, you can pass in a session_params object. Its settings
 		// will take effect before the session starts up.
+		// The overloads taking ``flags`` can be used to start a session in
+		// paused mode (by passing in ``session::paused``). Note that
+		// ``add_default_plugins`` do not have an affect on constructors that
+		// take a session_params object. It already contains the plugins to use.
 		explicit session(session_params const& params)
-		{ start(session_params(params), nullptr); }
+		{ start({}, session_params(params), nullptr); }
 		explicit session(session_params&& params)
-		{ start(std::move(params), nullptr); }
+		{ start({}, std::move(params), nullptr); }
+		explicit session(session_params const& params, session_flags_t const flags)
```

Comment:
Just to be sure... Is it impossible to implement it via session_params?

Reply:
no, and I could do that in `master`. But the layout of `session_params` is locked in the ABI for 1.2 and 2.0